src/build/specialize_types/IR.zig

```diff
@@ -125,129 +147,157 @@ pub const Expr = union(enum) {
 
     compiler_bug: Problem.Compiler,
 
-    /// todo
+    /// List of Expr
     pub const List = collections.SafeList(@This());
-    /// todo
+    /// Index into an Expr.List
     pub const Idx = List.Idx;
-    /// todo
+    /// Sublist reference of Expr.List
     pub const Slice = List.Slice;
-    /// todo
+    /// Sublist reference into Expr.List that is guaranteed to be non-empty
     pub const NonEmptySlice = List.NonEmptySlice;
-    /// todo
+    /// ??? Type annotated expression e.g. `1: U64` ?
     pub const Typed = struct {
```

Comment:
Like patterns, this is just a way to store the type solved earlier in typechecking with an expression if we need to know about the pair of them somewhere else in the specialize_types IR. This doesn't have to do with type annotations.